### PR TITLE
BUG: Fix divmod fill value, closes #26987

### LINKED ISSUE #26987: AttributeError with divmod(Series, Series) with 0 in the denominator
Strange one. This works,

```python
In [39]: a = pd.Series([1, 1])

In [40]: divmod(a, pd.Series([1, 2]));
```

This raises

```pytb
In [41]: divmod(a, pd.Series([0, 2]));
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-41-04277abe6f5a> in <module>
----> 1 divmod(a, pd.Series([0, 2]));

~/sandbox/pandas/pandas/core/ops.py in wrapper(left, right)
   1714             rvalues = rvalues.values
   1715
-> 1716         result = safe_na_op(lvalues, rvalues)
   1717         return construct_result(left, result,
   1718                                 index=left.index, name=res_name, dtype=None)

~/sandbox/pandas/pandas/core/ops.py in safe_na_op(lvalues, rvalues)
   1660         try:
   1661             with np.errstate(all='ignore'):
-> 1662                 return na_op(lvalues, rvalues)
   1663         except Exception:
   1664             if is_object_dtype(lvalues):

~/sandbox/pandas/pandas/core/ops.py in na_op(x, y)
   1640             result = masked_arith_op(x, y, op)
   1641
-> 1642         result = missing.fill_zeros(result, x, y, op_name, fill_zeros)
   1643         return result
   1644

~/sandbox/pandas/pandas/core/missing.py in fill_zeros(result, x, y, name, fill)
    544             mask = ((y == 0) & ~np.isnan(result)).ravel()
    545
--> 546             shape = result.shape
    547             result = result.astype('float64', copy=False).ravel()
    548

AttributeError: 'tuple' object has no attribute 'shape'
```

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #26987
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Takes over from #27130 by implementing dispatch_fill_zeros.  This is unfortunately similar to dispatch_missing, and eventually the two functions will be merged.  ATM I'm trying to do exactly that in the branch for 27130, having lots of trouble pop up with sparse and IntegerArray.

### CODE REVIEW

File: pandas/tests/arithmetic/test_numeric.py [hunk 1]
```diff
@@ -299,6 +299,54 @@ def test_ser_div_ser(self, dtype1, dtype2):
         tm.assert_series_equal(result, expected)
         assert not result.equals(second / first)
 
+    @pytest.mark.parametrize(
+        "dtype2",
+        [
+            np.int64,
+            np.int32,
```
Comment (jreback) [inline pandas/tests/arithmetic/test_numeric.py]: hmm, don;t' we have a fixture for this?

File: pandas/tests/arithmetic/test_numeric.py [hunk 2]
```diff
@@ -662,7 +710,9 @@ def test_modulo2(self):
             result2 = p["second"] % p["first"]
             assert not result.equals(result2)
 
-            # GH#9144
+    def test_modulo_zero_int(self):
+        # GH#9144
+        with np.errstate(all="ignore"):
```
Comment (jreback) [inline pandas/tests/arithmetic/test_numeric.py]: RuntimeWarning?

File: doc/source/whatsnew/v0.25.1.rst [hunk 3]
```diff
@@ -56,7 +56,7 @@ Timezones
 
 Numeric
 ^^^^^^^
-
+- Bug in ``divmod`` with a :class:`Series` object containing zeros incorrectly raising ``AttributeError`` (:issue:`26987`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.1.rst]: I think ok for 0.25.0

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/arithmetic/test_numeric.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/arithmetic/test_numeric.py]: This is broken off of the previous test, which uses the context in the same way.

File: pandas/tests/arithmetic/test_numeric.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/arithmetic/test_numeric.py]: Looks like we have `any_real_dtype`, includes `int` which is not included here.  I copied this list of dtypes from another test in this file.  I'll do a pass to use the fixture consistently in this file.

Comment (jreback) [reply pandas/tests/arithmetic/test_numeric.py]: great, yeah can do as a followup

File: pandas/tests/arithmetic/test_numeric.py (hunk 2 above)
Comment (jreback) [reply pandas/tests/arithmetic/test_numeric.py]: k

**Comment by jreback** [comment]
this supersedes #27130 right?

**Comment by jbrockmendel** [comment]
yes, but I'm keeping that open because that branch has evolved into addressing other bugs

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**